wallet/src/lib.rs

```diff
@@ -242,13 +250,140 @@ fn make_asset_certificate_header(asset_hashes: &AssetHashes, asset_name: &str) -
+        let replacement = "<script>var s=document.createElement('script');s.src=\"$name\";document.head.appendChild(s);</script>";
+        let processed = re.replace_all(index_html, replacement);
+        processed.to_string()
+    };
+    static ref INDEX_HTML_STR_HASH: [u8; 32] = {
+        let bytes = INDEX_HTML_STR.as_bytes();
+        let mut hasher = sha2::Sha256::new();
+        hasher.update(&bytes);
+        hasher.finalize().into()
+    };
+    static ref INDEX_HTML_JS_HASHES: String = {
+        let re = Regex::new("<script>(.*?)</script>").unwrap();
+        let mut res = String::new();
+        for cap in re.captures_iter(&*INDEX_HTML_STR) {
+            let s = &cap[1];
+            let hash = &sha2::Sha256::digest(s.as_bytes());
+            let hash = base64::encode(hash);
+            res.push_str(&format!("'sha256-{hash}' "));
+        }
+        res
+    };
+}
+
+/// List of recommended security headers as per https://owasp.org/www-project-secure-headers/
+/// These headers enable browser security features (like limit access to platform apis and set
+/// iFrame policies, etc.).
+fn security_headers() -> Vec<HeaderField> {
+    let hashes = INDEX_HTML_JS_HASHES.to_string();
+    vec![
+        ("X-Frame-Options".to_string(), "DENY".to_string()),
+        ("X-Content-Type-Options".to_string(), "nosniff".to_string()),
+        // Content Security Policy
+        //
+        // The sha256 hash matches the inline script in index.html. This inline script is a workaround
+        // for Firefox not supporting SRI (recommended here https://csp.withgoogle.com/docs/faq.html#static-content).
+        // This also prevents use of trusted-types. See https://bugzilla.mozilla.org/show_bug.cgi?id=1409200.
+        //
+        // script-src 'unsafe-eval' is required because agent-js uses a WebAssembly module for the
+        // validation of bls signatures.
+        // There is currently no other way to allow execution of WebAssembly modules with CSP.
+        // See https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md.
+        //
+        // style-src 'unsafe-inline' is currently required due to the way styles are handled by the
+        // application. Adding hashes would require a big restructuring of the application and build
+        // infrastructure.
+        //
+        // NOTE about `script-src`: we cannot use a normal script tag like this
+        //   <script src="index.js" integrity="sha256-..." defer></script>
+        // because Firefox does not support SRI with CSP: https://bugzilla.mozilla.org/show_bug.cgi?id=1409200
+        // Instead, we add it to the CSP policy
+        (
+            "Content-Security-Policy".to_string(),
+            format!(
+                "default-src 'none';\
+             connect-src 'self' https://ic0.app;\
+             img-src 'self' data:;\
+             script-src {} 'unsafe-eval' 'strict-dynamic' https:;\
```

Comment:
Sorry I recommended to remove 'unsafe-inline' here. But actually it is recommended to have it here. Could you add it back? 
* In CSP v3 this is ignored because of strict-dynamic. 
* In older browsers with older CSP versions this helps to make the site still work. 

FYI: You can see this using https://csp-evaluator.withgoogle.com/ and pasting the CSP there. 

```suggestion
             script-src {} 'unsafe-inline' 'unsafe-eval' 'strict-dynamic' https:;\
```